Support/ERCS_Logs/ERCS_AzureStackLogs.ps1

```diff
@@ -750,27 +1203,46 @@ if($IP)
 			Write-Host "`n `t[INFO] Saving AzureStackStampInformation to $($Env:SystemDrive)\$($sharename)" -ForegroundColor Green
 			$StampInformation | ConvertTo-Json | Out-File -FilePath "$($Env:SystemDrive)\$($sharename)\AzureStackStampInformation.json" -Force
 		}
-
-        # output for transcript
-        try
-        {
-        Write-Host "`n `t[INFO] Getting Azure Stack transcript" -ForegroundColor Green
-		Invoke-Command -Session $s -ScriptBlock {Close-PrivilegedEndpoint -TranscriptsPathDestination "\\127.0.0.1\C`$"}
-		$Transcript = Get-ChildItem -Path "\\$($IP)\C`$" | Where {($_.Name -like "Transcripts_*")} | sort -Descending -Property CreationTime | select -first 1
-		#copy the transcript to share 
-		Copy-Item -Path $Transcript.FullName -Destination $ShareINFO -Force
-		#test to make sure the transcript is copied
-		$localtranscript = Test-Path -Path "$($shareinfo)\$($Transcript.Name)"
-		if ($localtranscript -eq $true) {Remove-Item -Path $Transcript.FullName -Force}
-        }
-        catch [System.Management.Automation.RemoteException]
-        {
-            Write-Host "`n`t`t[Error] Exception caught: $_" -ForegroundColor Red
-        }
-        finally
-        {
-            Remove-PSSession $s
-        }
+		if($CheckADSK -ne 1)
+		{
+			# output for transcript mutinode
+			try
+			{
+			Write-Host "`n `t[INFO] Getting Azure Stack transcript" -ForegroundColor Green
+			Invoke-Command -Session $s -ScriptBlock {Close-PrivilegedEndpoint -TranscriptsPathDestination $using:ShareINFO -Credential $using:shareCred -ErrorAction SilentlyContinue -WarningAction SilentlyContinue} -ErrorAction SilentlyContinue -WarningAction SilentlyContinue
+			}
+			catch [System.Management.Automation.RemoteException]
+			{
+				Write-Host "`n`t`t[Error] Exception caught: $_" -ForegroundColor Red
+			}
+			finally
+			{
+				Remove-PSSession $s
+			}
+		}
+		else
+		{
+			Try
+			{
+			# output for transcript ASDK
+			Write-Host "`n `t[INFO] Getting Azure Stack transcript" -ForegroundColor Green
+			Invoke-Command -Session $s -ScriptBlock {Close-PrivilegedEndpoint -TranscriptsPathDestination "\\127.0.0.1\C`$" -WarningAction SilentlyContinue} -WarningAction SilentlyContinue
+			$Transcript = Get-ChildItem -Path "\\$($IP)\C`$" | Where {($_.Name -like "Transcripts_*")} | sort -Descending -Property CreationTime | select -first 1
+			#copy the transcript to share 
+			Copy-Item -Path $Transcript.FullName -Destination $ShareINFO -Force
+			#test to make sure the transcript is copied
+			$localtranscript = Test-Path -Path "$($shareinfo)\$($Transcript.Name)"
+			if ($localtranscript -eq $true) {Remove-Item -Path $Transcript.FullName -Force}
+			}
+			catch [System.Management.Automation.RemoteException]
+			{
+				Write-Host "`n`t`t[Error] Exception caught: $_" -ForegroundColor Red
+			}
+			finally
+			{
+				Remove-PSSession $s

```

Comment:
your removing the session here but it would be better to put the entire thing in a try finally for all the sessions.